src/components/NavBar.vue

```diff
@@ -0,0 +1,371 @@
+        >
+      </div>
+      <div class="pt-4 pb-3 border-t border-gray-200">
+        <div class="flex items-center px-4">
+        <template v-if="isUserAuth">
+          <div class="flex-shrink-0">
+            <img
+              class="h-10 w-10 rounded-full"
+              :src="this.user.photoURL"
+              alt=""
+            />
+          </div>
+          <div class="ml-3">
+            <div class="text-base font-medium text-gray-800">
+              {{ this.user.displayName }}
+            </div>
+            <div  class="text-sm font-medium text-gray-500">
+              {{ this.user.email }}
+            </div>
+        
+          </div>
+          </template>
+          <template v-else>
+          <div class="flex-shrink-0">
+            <img
+              class="h-10 w-10 rounded-full"
+              :src="require(`../assets/images/${imgURL}`)"
+              alt=""
+            />
+          </div>
+          </template>
+          <button
+            class="ml-auto flex-shrink-0 bg-white p-1 text-gray-400 rounded-full hover:text-gray-500 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
+          >
+            <span class="sr-only">View notifications</span>
+            <!-- Heroicon name: outline/bell -->
+            <svg
+              class="h-6 w-6"
+              xmlns="http://www.w3.org/2000/svg"
+              fill="none"
+              viewBox="0 0 24 24"
+              stroke="currentColor"
+              aria-hidden="true"
+            >
+              <path
+                stroke-linecap="round"
+                stroke-linejoin="round"
+                stroke-width="2"
+                d="M15 17h5l-1.405-1.405A2.032 2.032 0 0118 14.158V11a6.002 6.002 0 00-4-5.659V5a2 2 0 10-4 0v.341C7.67 6.165 6 8.388 6 11v3.159c0 .538-.214 1.055-.595 1.436L4 17h5m6 0v1a3 3 0 11-6 0v-1m6 0H9"
+              />
+            </svg>
+          </button>
+        </div>
+        <div v-if="isUserAuth" class="mt-3 space-y-1">
+          <a
+            href="#"
+            class="block px-4 py-2 text-base font-medium text-gray-500 hover:text-gray-800 hover:bg-gray-100"
+            >Your Profile</a
+          >
+          <a
+            href="#"
+            class="block px-4 py-2 text-base font-medium text-gray-500 hover:text-gray-800 hover:bg-gray-100"
+            >Settings</a
+          >
+          <div @click.stop.prevent="signOut">
+          <a
+            href="#"
+            class="block px-4 py-2 text-base font-medium text-gray-500 hover:text-gray-800 hover:bg-gray-100"
+            >Sign out</a
+          >
+          </div>
+        </div>
+        <div v-else class="mt-3 space-y-1" @click.stop.prevent="signInWithRedirect('google')"> 
+          <a
+            href="#"
+            class="block px-4 py-2 text-base font-medium text-gray-500 hover:text-gray-800 hover:bg-gray-100"
+            >Sign In</a
+          >
+        </div>
+      </div>
+    </div>
+  </nav>
```

Comment:
This is the TailwindUI HTML template, but they offer a Vue template as well. We should leverage the Vue code they have instead since it's already broken into expected components.

Reply:
Per Conversation, using HTML version as Tailwind Vue version has dependencies that do not work with Nuxt (Vue version 2.0) requires Vue 3.0.